File: springboot-web/json-web-examples/src/test/java/techtabu/json/mapkey/MapKeyControllerTest.java
package techtabu.json.mapkey;

import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

/**
 * @author TechTabu
 */

@Slf4j
@WebMvcTest
public class MapKeyControllerTest {

    @Autowired
    private MockMvc mockMvc;

    @Test
    public void testGetAll() throws Exception {
        log.info("Testing testGetAll");
        this.mockMvc.perform(get("/mapkey/"))
                .andDo(print())
                .andExpect(status().isOk());
    }

    @Test
    public void testGetAllBySerializer() throws Exception {
        log.info("Testing testGetAllBySerializer");
        this.mockMvc.perform(get("/mapkey/byserializer"))
                .andDo(print())
                .andExpect(status().isOk());
    }
}

File: springboot-web/json-web-examples/src/test/java/techtabu/json/mapkey/MapKeyControllerTest.kt
package techtabu.json.mapkey

import lombok.extern.slf4j.Slf4j
import mu.KotlinLogging
import org.junit.jupiter.api.Test
import org.junit.platform.commons.logging.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultHandlers
import org.springframework.test.web.servlet.result.MockMvcResultMatchers

/**
 * @author TechTabu
 */
@Slf4j
@WebMvcTest
class MapKeyControllerTest {

    private val log = KotlinLogging.logger{}

    @Autowired
    private val mockMvc: MockMvc? = null

    @Test
    @Throws(Exception::class)
    fun testGetAll() {
        log.info("Testing testGetAll")
        mockMvc!!.perform(MockMvcRequestBuilders.get("/mapkey/"))
            .andDo(MockMvcResultHandlers.print())
            .andExpect(MockMvcResultMatchers.status().isOk)
    }

    @Test
    @Throws(Exception::class)
    fun testGetAllBySerializer() {
        log.info("Testing testGetAllBySerializer")
        mockMvc!!.perform(MockMvcRequestBuilders.get("/mapkey/byserializer"))
            .andDo(MockMvcResultHandlers.print())
            .andExpect(MockMvcResultMatchers.status().isOk)
    }
}